gfw_pixetl/pixetl.py

```diff
@@ -41,20 +27,36 @@
     default=False,
     help="Overwrite existing tile in output location",
 )
+@click.argument("layer_json", type=str)
 def cli(
-    name: str,
-    version: str,
-    source_type: str,
-    field: str,
-    grid_name: str,
-    subset: Optional[List[str]],
-    overwrite: bool,
+    layer_json: str, subset: Optional[List[str]], overwrite: bool,
 ):
-    """NAME: Name of dataset"""
+    layer_def = LayerModel.parse_raw(layer_json)
 
-    tiles, skipped_tiles, failed_tiles = pixetl(
-        name, version, source_type, field, grid_name, subset, overwrite,
-    )
+    # Validate fields sooner rather than later
+    # On the other hand, moving this validation inside pixetl() would allow
+    # for easier testing...
+    if not utils.verify_version_pattern(layer_def.version):
+        message = "Version number does not match pattern"
+        LOGGER.error(message)
+        raise ValueError(message)
+
+    # Validate resampling_method values. I tried to do this with an enum
+    # (see commented-out lines in models.py and resampling.py) but couldn't
+    # get it to work
+    if layer_def.resampling not in resampling_methods.keys():
+        raise ValueError(f"Invalid resampling method specified: {layer_def.resampling}")
+
+    # Validate data_type values. Ideally turn this into an enum for Pydantic
+    # to validate automatically.
+    if layer_def.data_type not in data_types.keys():
+        raise ValueError(f"Invalid data_type specified: {layer_def.data_type}")
+
+    # Raster sources must have an source URI
+    if layer_def.source_type == "raster" and layer_def.uri is None:
+        raise ValueError("URI specification is required for raster sources")
+
+    tiles, skipped_tiles, failed_tiles = pixetl(layer_def, subset, overwrite,)
```

Comment:
This can be part of `LayerModel`

use REGEX for version pattern and emums to check if resampling and data_type values are valid.

Pydantic also supports custom validators. If you want to validate combination of two attributes you can add a root validator:

```python
@root_validator
def check_raster_uri(cls, values):
    source_type, uri = values.get('source_type'), values.get('uri')
    if source_type == "raster" and uri is None:
        raise ValueError("URI specification is required for raster sources")        
    return values
```